### PR TITLE
Create mutations

### PR DESCRIPTION
**Pull request by rotimi-babalola**
Create mutations to
- add an expense
- edit an expense
- delete an expense

### CODE REVIEW

File: server/schemas/mutations/delete-expense.mutation.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  args: {
+    expenseId: { type: new GraphQLNonNull(GraphQLString) },
+  },
+  resolve: async (obj, { expenseId }) => {
+    try {
+      const deleted = await Expense.findByIdAndRemove(expenseId).exec();
+      if (deleted) {
+        return 'Successfully deleted!';
+      }
+      throw new Error('Unable to delete');
```
Comment (ghost) [inline server/schemas/mutations/delete-expense.mutation.js:19]: Hmmm. I see what you are trying to do, however, the error might be handled in the catch block. No?

Have you tested to find that you get the 'Unable to delete' message?

File: server/schemas/mutations/edit-expense.mutation.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  name: 'EditExpense',
+  fields: {
+    name: { type: GraphQLString },
+    category: { type: CategoryType },
+    amount: { type: GraphQLFloat },
+  },
+});
+
+export default {
+  type: new GraphQLNonNull(ExpenseType),
```
Comment (ghost) [inline server/schemas/mutations/edit-expense.mutation.js:22]: These `non nulls`, is it possible to simply do an `ExpenseType!`, with `!` signifying required?

File: server/schemas/mutations/edit-expense.mutation.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    fieldsToUpdate: { type: new GraphQLNonNull(EditExpenseType) },
+  },
+  resolve: async (_, { expenseId, fieldsToUpdate }) => {
+    try {
+      const updated = await Expense.findByIdAndUpdate(expenseId, fieldsToUpdate, { new: true }).exec();
+      // ensure updated is not null or undefined
+      if (updated) {
+        return updated;
+      }
+      throw new Error('Unable to update');
```
Comment (ghost) [inline server/schemas/mutations/edit-expense.mutation.js:34]: Same as in the previous file.

File: server/schemas/mutations/delete-expense.mutation.js (hunk 1 above)
Comment (rotimi-babalola) [reply server/schemas/mutations/delete-expense.mutation.js:19]: Yep. I've been able to verify this locally. 

Basically, it throws this error when you try a `findById` operation for an ID that does not exist in the DB. MongoDB (or Mongoose) returns `null`. I don't want it to return `null` so I make it throw this error @oreoluwade

File: server/schemas/mutations/edit-expense.mutation.js (hunk 2 above)
Comment (rotimi-babalola) [reply server/schemas/mutations/edit-expense.mutation.js:22]: Hmmm... 🤔

File: server/schemas/mutations/delete-expense.mutation.js (hunk 1 above)
Comment (ghost) [reply server/schemas/mutations/delete-expense.mutation.js:19]: Awesome!

File: server/schemas/mutations/edit-expense.mutation.js (hunk 2 above)
Comment (rotimi-babalola) [reply server/schemas/mutations/edit-expense.mutation.js:22]: Doesn't work 😢😞

File: server/schemas/mutations/edit-expense.mutation.js (hunk 3 above)
Comment (ghost) [reply server/schemas/mutations/edit-expense.mutation.js:34]: This concern has been addressed. 🍡

File: server/schemas/mutations/edit-expense.mutation.js (hunk 2 above)
Comment (ghost) [reply server/schemas/mutations/edit-expense.mutation.js:22]: Haha. It's fine then. Used to do it that way while using graphene in python, that's why I was wondering.

**Review by ghost — APPROVED** [review]

**Pull request merged**